### PR TITLE
Return HEVC, VP8, VP9 HW decoders back to library

### PR DESCRIPTION
**Pull request by onabiull**
Library is a better place for pure HW codecs.
This commit provides stub implementation for backward compatibility.

Co-authored-by: Gurulev, Dmitry <dmitry.gurulev@intel.com>
Co-authored-by: Nabiullin, Oleg <oleg.nabiullin@intel.com>

### CODE REVIEW

File: _studio/shared/include/mfx_config.h [hunk 1]
```diff
@@ -50,25 +50,21 @@
 
 #if !defined(LINUX_TARGET_PLATFORM) || defined(LINUX_TARGET_PLATFORM_BDW) || defined(LINUX_TARGET_PLATFORM_CFL) || defined(LINUX_TARGET_PLATFORM_BXT)
     #if !defined(ANDROID)
-        // h264d
+        // HW decodrs are part of library
```
Comment (lakulako) [inline _studio/shared/include/mfx_config.h]: "decod[e]rs"

File: _studio/shared/include/mfx_config.h [hunk 2]
```diff
@@ -50,25 +50,21 @@
...
-
+        #define MFX_ENABLE_VP9_VIDEO_DECODE_HW
+        
         //h264e
         #define MFX_ENABLE_H264_VIDEO_ENCODE
 
         //h265e
-        #if defined(AS_HEVCD_PLUGIN) || defined(AS_HEVCE_PLUGIN) || defined(MFX_VA)
+        #if defined(AS_HEVCE_PLUGIN) || defined(MFX_VA)
+            #define MFX_ENABLE_H265_VIDEO_DECODE
```
Comment (dvrogozh) [inline _studio/shared/include/mfx_config.h]: This is inconsistent with `#if defined(AS_HEVCE_PLUGIN)`

File: _studio/mfx_lib/plugin/include/mfx_vp9_dec_plugin.h [hunk 3]
```diff
@@ -21,126 +21,23 @@
 #if !defined(__MFX_VP9_DEC_PLUGIN_INCLUDED__)
```
Comment (dvrogozh) [inline _studio/mfx_lib/plugin/include/mfx_vp9_dec_plugin.h]: Common practice to name such thing is __MFX_VP9_DEC_PLUGIN_H__, INCLUDED looks ugly

File: _studio/mfx_lib/CMakeLists.txt [hunk 4]
```diff
@@ -89,7 +89,7 @@ include_directories( ${MSDK_STUDIO_ROOT}/shared/asc/include )
 include_directories( ${MSDK_STUDIO_ROOT}/hevce_hw/h265/include )
 
 list( APPEND cdirs
-  brc h264_enc mpeg2_dec  vc1_dec vc1_common
+  brc h264_enc mpeg2_dec vc1_dec vp9_dec vc1_common
```
Comment (dvrogozh) [inline _studio/mfx_lib/CMakeLists.txt]: why there is no h264_dec and h265_dec in this list? and if we don't need these, why we need vc1 and vc9

Comment (onabiull) [reply _studio/mfx_lib/CMakeLists.txt]: Unfortunately all of them currently required for compilation. Refactoring out of scope for this PR.

File: _studio/shared/include/mfx_config.h (hunk 2 above)
Comment (onabiull) [reply _studio/shared/include/mfx_config.h]: Will fix.

File: _studio/mfx_lib/plugin/include/mfx_vp9_dec_plugin.h (hunk 3 above)
Comment (onabiull) [reply _studio/mfx_lib/plugin/include/mfx_vp9_dec_plugin.h]: Refactoring out of scope for this PR. Will consider alignment of include guards later.

**Pull request merged**